### PR TITLE
Prevent duplicate messages for orders

### PR DESCRIPTION
**Pull request by avstudnitz**

### CODE REVIEW

File: Model/ProcessedOrdersRegistry.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        $this->processedOrderIdentifiers[$this->getIdentifier($order)] = true;
+    }
+
+    public function isOrderProcessed(OrderInterface $order): bool
+    {
+        $identifier = $this->getIdentifier($order);
+
+        if (isset($this->processedOrderIdentifiers[$identifier])) {
+            unset ($this->processedOrderIdentifiers[$identifier]);
```
Comment (gowrizrh) [inline Model/ProcessedOrdersRegistry.php]: What's the reason behind marking it as not processed after the first time?

File: Model/ProcessedOrdersRegistry.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Sales\Api\Data\OrderInterface;
+
+/**
+ * Save a list of already processed orders during this request.
+ */
+class ProcessedOrdersRegistry
+{
+    const FIELD_NAME_IDENTIFIER = 'entity_id';
```
Comment (gowrizrh) [inline Model/ProcessedOrdersRegistry.php]: Is this being used or could this be cleaned up?

**Review by gowrizrh — COMMENTED** [review]
just a couple questions

File: Model/ProcessedOrdersRegistry.php (hunk 2 above)
Comment (avstudnitz) [reply Model/ProcessedOrdersRegistry.php]: You're right, can be cleaned up.

File: Model/ProcessedOrdersRegistry.php (hunk 1 above)
Comment (avstudnitz) [reply Model/ProcessedOrdersRegistry.php]: Good question. I was taking it over from the customer registry, but don't remember the reason for that.

**Review by gowrizrh — APPROVED** [review]
Wouldn't this prevent order updates from propogating through? If the order is new and is saved in the registry and the next time the order is updated becase it's paid or status updated then this effectively would block it because the code sees it as it's already processed right?

I've approved but you may want to test those cases to see if it's working as intended for you.

**Comment by avstudnitz** [comment]
> Wouldn't this prevent order updates from propogating through? If the order is new and is saved in the registry and the next time the order is updated becase it's paid or status updated then this effectively would block it because the code sees it as it's already processed right?
> 
> I've approved but you may want to test those cases to see if it's working as intended for you.

@gowrizrh That would be in a different request then - the registry is cleared after each request and not persistant. In theory, several changes to an order could happen in a single request, but that would be very unusual (and also wouldn't make much sense IMO).

**Comment by gowrizrh** [comment]
makes sense 👍

**Pull request merged**